### ISSUE TITLE
F841 is triggered by unused variables in tuple unpacking from with statements

### ISSUE THREAD

**Issue by fnep** [charliermarsh/ruff#890]
When tuple unpacking is used within `as` of a with statement, F841 is raised for unused variables. This is fine if really all variables are unused, but if I need only one, I have not other choice then getting all. This happens only within functions.

```python3
def connect():
    return (None, None)


def test():
    with connect() as (connection, cursor):
        cursor.execute("SELECT * FROM users")

```

using ruff 0.0.135 gives me
 
```
test.py:7:24: F841 Local variable `connection` is assigned to but never used
```

In flake8, this is case works.

**Comment by JonathanPlasse** [comment]
You can use an underscore to ignore unused values.

```python
def connect():
    return (None, None)


def test():
    with connect() as (_, cursor):
        cursor.execute("SELECT * FROM users")
```

**Comment by fnep** [comment]
As a workaround for a few places `_` would work, I'm testing a huge codebase. Thank you anyway for the idea.

Neither pyflakes nor flake8 find this an issue in my tests.

**Comment by JonathanPlasse** [comment]
> Flakes should have detected it.

I meant that it might be a bug.

**Comment by charliermarsh** [comment]
Hmm, yeah, I think we may want to allow this.

Some related stuff here: https://github.com/charliermarsh/ruff/issues/161.

**Comment by charliermarsh** [comment]
(Will be fixed today.)

**Comment by fnep** [comment]
Works perfect in ruff==0.0.137, thank you :)

**Comment by charliermarsh** [comment]
Awesome :)

**Comment by merlynmarc** [comment]
I would prefer that ruff flags the unused variable (or at least gives me the option to).  I would write the original code as the underscore version (actually as _connection).  The fact that ruff doesn't flag this is a blocker for me adopting ruff over pylint.
